src/components/forms/types.ts

```diff
@@ -66,7 +65,10 @@ export interface VolunteerSpecificAppData {
 export type ApplicationData = HackerApplicationData &
 	HackerSpecificAppData &
 	MentorSpecificAppData &
-	VolunteerSpecificAppData;
+	VolunteerSpecificAppData &{
+		rsvp?: boolean;
+		applicationStatus: "pending" | "rejected" | "accepted";
+	};
```

Comment:
You could create a new type called `GenericApplicationData` that has both rsvp and applicationStatus. This way it would be clearer to read.